lightly/api/bitmask.py

```diff
@@ -0,0 +1,216 @@
+""" Download from Lightly API """
+
+# Copyright (c) 2020. Lightly AG and its affiliates.
+# All Rights Reserved
+
+from typing import List
+
+
+def _hex_to_int(hexstring: str) -> int:
+    """Converts a hex string representation of an integer to an integer.
+
+    """
+    return int(hexstring, 16)
+
+
+def _bin_to_int(binstring: str) -> int:
+    """Converts a binary string representation of an integer to an integer.
+
+    """
+    return int(binstring, 2)
+
+
+def _int_to_hex(x: int) -> str:
+    """Converts an integer to a hex string representation.
+
+    """
+    return hex(x)
+
+def _int_to_bin(x: int) -> str:
+    """Converts an integer to a binary string representation.
+
+    """
+    return bin(x)
+
+def _get_nonzero_bits(x: int) -> List[int]:
+    """Returns a list of indices of nonzero bits in x.
+
+    """
+    offset = 0
+    nonzero_bit_indices = []
+    while x > 0:
+        # if the number is odd, there is a nonzero bit at offset
+        if x % 2 > 0:
+            nonzero_bit_indices.append(offset)
+        # increment the offset and divide the number x by two (rounding down)
+        offset += 1
+        x = x // 2
+    return nonzero_bit_indices
+
+def _invert(x: int) -> int:
+    """Flips every bit of x as if x was an unsigned integer.
+
+    """
+    # use XOR of x and 0xFFFFFF to get the inverse
+    # return x ^ (2 ** (x.bit_length()) - 1)
+    # TODO: the solution above can give wrong answers for the case where
+    # the tag representation starts with a zero, therefore it needs to know
+    # the exact number of samples in the dataset to do a correct inverse
+    raise NotImplementedError('This method is not implemented yet...')
+
+def _union(x: int, y: int) -> int:
+    """Uses bitwise OR to get the union of the two masks.
+
+    """
+    return x | y
+
+
+def _intersection(x: int, y: int) -> int:
+    """Uses bitwise AND to get the intersection of the two masks.
+
+    """
+    return x & y
+
+
+def _get_kth_bit(x: int, k: int) -> int:
+    """Returns the kth bit in the mask from the right.
+
+    """
+    mask = 1 << k
+    return x & mask
+
+
+def _set_kth_bit(x: int, k: int) -> int:
+    """Sets the kth bit in the mask from the right.
+
+    """
+    mask = 1 << k
+    return x | mask
+
+
+def _unset_kth_bit(x: int, k: int) -> int:
+    """Clears the kth bit in the mask from the right.
+
+    """
+    mask = ~(1 << k)
+    return x & mask
+
+
+class BitMask:
+    """Utility class to represent and manipulate tags.
+
+    Attributes:
+        x:
+            An integer representation of the binary mask.
+
+    Examples:
+        >>> # the following are equivalent
+        >>> mask = BitMask(6)
+        >>> mask = BitMask.from_hex('0x6')
+        >>> mask = Bitmask.from_bin('0b0110')
```

Comment:
Please add an example covering the data as it would come from the API:`"A8B42"` or so) `0x..` notation is Python internal once it's a hex string. But input to the BitMask is a normal string. 
Also, add an example to get the string (same format as API).

Reply:
I requested a change on your PR to use the `0x` notation and @japrescott agreed with me. Let's agree first and then make changes.

Reply:
Now, you can add an example :)